### PR TITLE
fix: selection range issue

### PR DESCRIPTION
**Pull request by kapral18**
based on a good visual select helper from https://github.com/ibhagwan/fzf-lua/blob/main/lua/fzf-lua/utils.lua#L533

additionally:

refactored a tiny bit with native base64. Yields exact same result for me as your custom helper.

### CODE REVIEW

File: lua/oatmeal.lua [hunk 1]
```diff
@@ -1,56 +1,32 @@
...
+    start_line = selection.start.line
+    end_line = selection["end"].line
   end
 
   session_context = {
     file_path = vim.fn.expand("%:p"),
     language = vim.bo.filetype,
     buffer_id = vim.fn.bufnr(),
-    code = to_base64(code),
+    code = vim.base64.encode(code),
```
Comment (dustinblackman) [inline lua/oatmeal.lua:29]: ❤️ Love this, I remember searching for it and not finding it. Thanks!

File: lua/utils.lua [hunk 2]
```diff
@@ -0,0 +1,60 @@
+function M.tbl_length(T)
```
Comment (dustinblackman) [inline lua/utils.lua]: 🚓 Unfortunately fzf-lua is licensed under AGPL, we can't just copy/paste it. :( I'd love to hear more about the issues you were having though, and possibly look for another solution. 

Possible alternative approach that could be mended together: https://github.com/neovim/neovim/pull/13896#issuecomment-1621702052

**Review by dustinblackman — CHANGES_REQUESTED** [review]

File: lua/oatmeal.lua (hunk 1 above)
Comment (dustinblackman) [inline lua/oatmeal.lua:29]: Love this! I remember looking for it, and not finding it.

Comment (dustinblackman) [inline lua/oatmeal.lua]: This isn't on Neovim stable yet. :(

File: lua/utils.lua (hunk 2 above)
Comment (kapral18) [reply lua/utils.lua]: no idea how exactly copyright works, but we can rename the variables, I don't think he has trademark for the word get_visual_selection or use of the word csrow or any nvim api used in this way or if conditions with vim mode combination tho... That would be absolutely insane if that's true, most open source would be unusable. I mean every time you write some snippet of code or `if vim.fn.mod()` you'd need to double check if there is a project with the same code. I mean you would quickly run out of code you can write using open apis, that is not copyrighted no ?

Comment (dustinblackman) [reply lua/utils.lua]: I'm no expert either, so I tend to be cautious. It's why I find myself not even touching projects that don't have licensing that's compatible with MIT when building new stuff. I know I'm totally over cautious, but I'm a rather be safe than sorry kind of guy.

I agree simple stuff like `if vim.fn.mod` no one is looking for or can rightfully own, but full modules I would think fall under it.

File: lua/oatmeal.lua [hunk 3]
```diff
@@ -1,28 +1,48 @@
 local base64 = require("base64")
+local utils = require("utils")
+
 local M = {}
 
 local config = {}
 local session_context = {}
 
+local function to_base64(data)
```
Comment (dustinblackman) [inline lua/oatmeal.lua:9]: I've changed this, no need to recommit

Comment (dustinblackman) [inline lua/oatmeal.lua:9]: I've changed this, no need to commit back in.

File: lua/utils.lua [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+function M.get_visual()
+  local start_pos = vim.fn.getpos("v")
+  local end_pos = vim.fn.getpos(".")
+
+  if start_pos == nil or end_pos == nil then
+    return ""
+  end
+
+  local start_row, start_col = start_pos[2], start_pos[3]
+  local end_row, end_col = end_pos[2], end_pos[3]
```
Comment (dustinblackman) [inline lua/utils.lua:12]: Do we need columns?

**Review by dustinblackman — APPROVED** [review]
I see you problem now, and this is a great fix. Thanks for handling all the comments!

**Pull request merged**